server/src/test/java/org/opensearch/index/store/StoreTests.java

```diff
@@ -1257,6 +1263,54 @@ public void testSegmentReplicationDiff() {
         assertTrue(diff.identical.isEmpty());
     }
 
+    @SuppressForbidden(reason = "sets the SEARCHABLE_SNAPSHOT_EXTENDED_COMPATIBILITY feature flag")
+    public void testReadSegmentsFromOldIndices() throws IOException {
+        int expectedIndexCreatedVersionMajor = SEARCHABLE_SNAPSHOT_EXTENDED_COMPATIBILITY_MINIMUM_VERSION.luceneVersion.major;
+        final String pathToTestIndex = "/indices/bwc/es-6.3.0/testIndex-es-6.3.0.zip";
+        Path tmp = createTempDir();
+        TestUtil.unzip(getClass().getResourceAsStream(pathToTestIndex), tmp);
+        final ShardId shardId = new ShardId("index", "_na_", 1);
+        Store store = null;
+        try {
+            AccessController.doPrivileged(
+                (PrivilegedAction<String>) () -> System.setProperty(FeatureFlags.SEARCHABLE_SNAPSHOT_EXTENDED_COMPATIBILITY, "true")
+            );
```

Comment:
Can you make this a bit more concise by creating an AutoCloseable helper class? The syntax would then look like:

```
try (FeatureFlagSetter setter = FeatureFlagSetter.set(FeatureFlags.SEARCHABLE_SNAPSHOT_EXTENDED_COMPATIBILITY)) {
    ... test impl ...
}
```

The class would be:


```
public final class FeatureFlagSetter implements AutoCloseable {
    private final String flag;

    private FeatureFlagSetter(String flag) {
        this.flag = flag;
    }

    public static FeatureFlagSetter set(String flag) {
        AccessController.doPrivileged((PrivilegedAction<String>) () -> System.setProperty(flag, "true"));
        return new FeatureFlagSetter(flag);
    }

    @Override
    public void close() throws Exception {
        AccessController.doPrivileged((PrivilegedAction<String>) () -> System.clearProperty(flag));
    }
}
```